src/main.rs

```diff
@@ -322,3 +337,64 @@ fn apply_suggestion(suggestion: &Suggestion) -> Result<(), ProgramError> {
 
     Ok(())
 }
+
+
+/// We need some special care for use suggestions,
+/// to eliminate duplicates, for example for HashMap,
+/// rustc suggests 4 variants:
```

Comment:
We should not be eliminating suggestions. If there are exact duplicates that's fine, but if we'd eliminate the duplicate for something like `Iter`, we're basically guaranteed to get the wrong one.